trl/rewards/accuracy_rewards.py

```diff
@@ -70,10 +74,119 @@ def accuracy_reward(completions: list[list[dict[str, str]]], solution: list[str]
                 extraction_mode="first_match",
             )
             # Compute binary rewards if verifiable, `None` otherwise to skip this example
-            reward = float(verify(gold_parsed, answer_parsed))
+            try:
+                reward = float(verify(gold_parsed, answer_parsed))
+            except Exception as e:
+                logger.warning(f"Verification failed with error: {e}. Assigning reward=None for this example.")
+                reward = None
```

Comment:
do you have an example that would raise an exception here 

```python
float(verify(gold_parsed, answer_parsed))
```

?

Since both are parsed, I don't think it can fail.

Reply:
Good point, I was following some old logic we had in `open-r1`. I checked a few edge cases and indeed even if the parser returns `[]` for the `answer_parsed` we still get 0 reward and no exception. One can trigger the exception with `raise_on_error=True` inside `verify`, but that doesn't seem super helpful for RL